### PR TITLE
Register form styled to match Login form, updated Login form placeholders

### PR DESCRIPTION
**Pull request by philer**
This should work. Translations are still missing, I dunno how that works. (By that I mean the actual translations. The `|trans` calls are in place).
Let me know if I did something wrong.

### CODE REVIEW

File: src/module/Ui/templates/authentication/login.twig [hunk 1]
```diff
@@ -14,13 +14,13 @@
 <div class="form-group">
     <div class="input-group">
         <span class="input-group-addon"><i class="fa fa-user"></i></span>
-        {{ formElement(form.get('email')) }}
+        {{ formElement(form.get('email').setAttribute('placeholder', 'Email'|trans)) }}
```
Comment (arekkas) [inline src/module/Ui/templates/authentication/login.twig]: you can set the placeholder attribute in the form class

File: src/module/Ui/templates/authentication/login.twig [hunk 2]
```diff
@@ -14,13 +14,13 @@
...
     </div>
 </div>
 <div class="form-group">
     <div class="input-group">
         <span class="input-group-addon"><i class="fa fa-lock"></i></span>
-        {{ formElement(form.get('password')) }}
+        {{ formElement(form.get('password').setAttribute('placeholder', 'Password'|trans)) }}
```
Comment (arekkas) [inline src/module/Ui/templates/authentication/login.twig]: see above

File: src/module/User/src/User/Form/Login.php [hunk 3]
```diff
@@ -27,12 +27,23 @@ public function __construct()
         $filter = new InputFilter();
         $this->setInputFilter($filter);
 
-        $this->add((new Text('email'))->setLabel('Email address:')->setAttribute('placeholder', 'Email'));
-        $this->add((new Password('password'))->setLabel('Password:')->setAttribute('placeholder', 'Password'));
-        $this->add((new Checkbox('remember'))->setLabel('Remember me')->setChecked(true));
+        $this->add((new Text('email'))
+            ->setAttribute('type','email')
```
Comment (arekkas) [inline src/module/User/src/User/Form/Login.php]: plase format correctly: `'type', 'email'`

File: src/module/User/src/User/Form/Login.php [hunk 4]
```diff
@@ -27,12 +27,23 @@ public function __construct()
...
+            ->setLabel('Email address:')
+            ->setAttribute('required','required')
+        );
+        $this->add((new Password('password'))
+            ->setLabel('Password:')
+            ->setAttribute('required','required')
```
Comment (arekkas) [inline src/module/User/src/User/Form/Login.php]: see above

File: src/module/Ui/templates/user/user/register.twig [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{#<div class="flasher">
```
Comment (arekkas) [inline src/module/Ui/templates/user/user/register.twig]: you can safely remove this

File: src/module/Ui/templates/user/user/register.twig [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    </div>
+</fieldset>
+<div class="form-group{{ tosErrors ? ' has-error' }}">
+    {{ formElement(form.get('tos')) }}
+    {{ tosErrors }}
+</div>
+<div class="form-group">
+    {{ formElement(form.get('submit')) }}
+</div>
+{{ form().closeTag(form) }}
```
Comment (arekkas) [inline src/module/Ui/templates/user/user/register.twig]: Add a newline at the end

File: src/module/Ui/templates/user/user/register.twig [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        <div class="input-group">
+            <span class="input-group-addon"><i class="fa fa-envelope"></i></span>
+            {{ formElement(form.get('email').setAttribute('placeholder', 'Email'|trans)) }}
+        </div>
+        {{ emailErrors }}
+    </div>
+    <div class="form-group{{ emailConfirmErrors ? ' has-error' }}">
+        <div class="input-group">
+            <span class="input-group-addon"><i class="fa fa-envelope"></i></span>
+            {{ formElement(form.get('emailConfirm').setAttribute('placeholder', 'Confirm email'|trans)) }}
```
Comment (arekkas) [inline src/module/Ui/templates/user/user/register.twig]: Please set the attributes in the form class

File: src/module/Ui/templates/authentication/login.twig (hunk 1 above)
Comment (philer) [inline src/module/Ui/templates/authentication/login.twig]: That's where they were, but they didn't get translated. Is there a way to apply translation immediately in the class?

Comment (arekkas) [inline src/module/Ui/templates/authentication/login.twig]: Ah yes I know that problem, there is a helper function for that: https://github.com/serlo-org/athene2/blob/master/src/public/index.php#L34

So just do `setAttribute('a', t('b'))`

**Comment by philer** [comment]
Okay so I moved the placeholder attributes to the models and then injected a $translator from the controllers. Since UserController has a getter for Form models it became a little ugly (I was tempted to use a switch statement). Is this all right or is there a better way to do it?

File: src/module/User/src/User/Form/Login.php [hunk 8]
```diff
@@ -19,7 +19,7 @@
 class Login extends Form
 {
 
-    public function __construct()
+    public function __construct($translator)
```
Comment (arekkas) [inline src/module/User/src/User/Form/Login.php]: Typehint `$translator`, should be something like Zend\Translator\Translator

File: src/module/Ui/templates/user/user/register.twig (hunk 6 above)
Comment (philer) [inline src/module/Ui/templates/user/user/register.twig]: fixed in ededa83, github seems to ignore it.

File: src/module/User/src/User/Form/Register.php [hunk 9]
```diff
@@ -19,7 +19,7 @@
 class Register extends Form
 {
 
-    public function __construct($entityManager)
+    public function __construct($entityManager, $translator)
```
Comment (arekkas) [inline src/module/User/src/User/Form/Register.php]: Typehint both arguments :) First should be Doctrine\ORM\EntityManager iirc

File: src/module/User/src/User/Controller/UserController.php [hunk 10]
```diff
@@ -104,8 +104,17 @@ public function getForm($name)
...
             $form = $this->forms[$name];
-            if ($name == 'register' || $name = 'settings') {
+            if ($name == 'register') {
+                
+                $this->forms[$name] = new $form(
+                    $this->getUserManager()->getObjectManager(),
+                    $this->getServiceLocator()->get('MvcTranslator')
+                );
+                
+            } else if ($name = 'settings') {
```
Comment (arekkas) [inline src/module/User/src/User/Controller/UserController.php]: you're assigning, not checking. Please use `==` or `===`

File: src/module/User/src/User/Controller/UserController.php [hunk 11]
```diff
@@ -104,8 +104,17 @@ public function getForm($name)
...
                 $this->forms[$name] = new $form($this->getUserManager()->getObjectManager());
+            } else if ($name = 'login') {
```
Comment (arekkas) [inline src/module/User/src/User/Controller/UserController.php]: see above

File: src/module/User/src/User/Controller/UserController.php [hunk 12]
```diff
@@ -104,8 +104,17 @@ public function getForm($name)
     {
         if (!is_object($this->forms[$name])) {
             $form = $this->forms[$name];
-            if ($name == 'register' || $name = 'settings') {
+            if ($name == 'register') {
+
```
Comment (arekkas) [inline src/module/User/src/User/Controller/UserController.php]: remove blank lines

**Comment by arekkas** [comment]
Looking good to me! Yep, that's the way to do it (injecting the translator) :) If you saw my previous comment (I already deleted it) with `t()` please ignore it, I mixed up something.

If/else is fine :)

File: src/module/User/src/User/Controller/UserController.php (hunk 10 above)
Comment (philer) [inline src/module/User/src/User/Controller/UserController.php]: oh dear god

File: src/module/Ui/templates/user/user/register.twig (hunk 6 above)
Comment (arekkas) [inline src/module/Ui/templates/user/user/register.twig]: Jep :)

File: src/module/User/src/User/Controller/UserController.php (hunk 10 above)
Comment (arekkas) [inline src/module/User/src/User/Controller/UserController.php]: Yeah, originally my mistake but while you're at it.. ;)

**Comment by arekkas** [comment]
Thanks!

**Pull request merged**